tests/test_pyth_client.py

```diff
@@ -0,0 +1,378 @@
+    mapping_account: PythMappingAccount,
+    mapping_account_entries: List[SolanaPublicKey],
+    product_account: PythProductAccount,
+    price_account: PythPriceAccount,
+    product_account_bytes: bytes,
+    price_account_bytes: bytes
+) -> None:
+    products = await pyth_client.get_products()
+    for product in products:
+        product.update_from(buffer=product_account_bytes, version=_VERSION_2)
+        prices = await product.get_prices()
+        for price in prices.values():
+            price.update_from(buffer=price_account_bytes, version=_VERSION_2)
+
+    accounts = await pyth_client.get_all_accounts()
+    for account in accounts:
+        if isinstance(account, PythMappingAccount):
+            assert account.key == mapping_account.key
+            assert account.entries == mapping_account_entries
+        elif isinstance(account, PythProductAccount):
+            assert dict(account) == dict(product_account)
+        elif isinstance(account, PythPriceAccount):
+            assert account.key == price_account.key
+            assert account.price_type == PythPriceType.PRICE
+            assert account.exponent == -9
+            assert account.num_components == 27
+            assert len(account.price_components) == account.num_components
+            assert account.last_slot == 96878111
+            assert account.valid_slot == 96878110
+            assert account.product_account_key == product_account.key
+            assert account.next_price_account_key is None
+
+
+@ pytest.mark.asyncio
+async def test_refresh_all_prices(
+    pyth_client: PythClient,
+    mock_get_account_info: AsyncMock,
+    mock_get_program_accounts: AsyncMock,
+    product_account: PythProductAccount,
+    price_account: PythPriceAccount
+) -> None:
+    await pyth_client.refresh_all_prices()
+    for product in pyth_client.products:
+        account = product.prices[PythPriceType.PRICE]
+        assert account.key == price_account.key
+        assert account.price_type == PythPriceType.PRICE
+        assert account.exponent == -9
+        assert account.num_components == 27
+        assert len(account.price_components) == account.num_components
+        assert account.last_slot == 96878111
+        assert account.valid_slot == 96878110
+        assert account.product_account_key == product_account.key
+        assert account.next_price_account_key is None
+
+
+@ pytest.mark.asyncio
+async def test_refresh_all_prices_no_program_key(
+    pyth_client_no_program_key: PythClient,
+    mock_get_account_info: AsyncMock,
+) -> None:
+    await pyth_client_no_program_key.refresh_all_prices()
+    for product in pyth_client_no_program_key.products:
+        with pytest.raises(NotLoadedException):
+            product.prices
+
+
+def test_create_watch_session(
+    pyth_client: PythClient,
+    watch_session: WatchSession
+) -> None:
+    ws = pyth_client.create_watch_session()
+    assert ws._next_subid() == watch_session._next_subid()
```

Comment:
What is this actually supposed to be testing? Is it possible to setup something so you test the public methods here?

This definitely expands the coverage, but I'm not sure what this is actually asserting other than the two subscription ids are the same.

Reply:
the purpose is to purely test the `create_watch_session()` function, with regards to the actual `WatchSession` functions, it could be done on a separate file?

Reply:
maybe for this I can simply test if it returns an instance of `WatchSession`?

Reply:
Ah ok, yeah that makes sense. The isinstance thing is a good idea here. I just wasn't quite sure what this was supposed to be doing, thanks for clarifying.

As an example, for the binary tests in #3, I was doing 1 test file for each of the big classes in pythclient.pythaccounts instead of 1 test file per source file. It makes it a little more organized that way. I try to make smaller test files when feasible.

Reply:
yup, I'm following your style but the `create_watch_session()` is in the `PythClient` class and there's a separate `WatchSession` class although both of them are in `pythclient.py`